### ISSUE TITLE
🐛 [BUG] Cache error in mas info/outdated

### ISSUE THREAD

**Issue by lonelykid** [mas-cli/mas#328]
## Your Environment

- `mas version`:1.8.1
- macOS version (`system_profiler SPSoftwareDataType -detailLevel mini`): 11.2.3

### mas Install Method

- [x] `brew install mas` (homebrew-core)
- [] [mas-cli/tap](https://github.com/mas-cli/homebrew-tap)
- [] `.pkg` installer from [releases](https://github.com/mas-cli/mas/releases)
- [] Built from source
   - Fork/branch: ? (e.g. mas-cli/master)
   - Xcode version: 10.?

## Describe the Bug

> A clear and concise description of what the bug is.

Running `mas info xx` or `mas outdated` sometimes incur errors like this
```
2021-03-16 21:23:54.978 mas[19064:1993163] INFO: fetch-response is unable to open the file /Users/username/Library/Caches/com.mphys.mas-cli/fsCachedData/9C5B8520-0F28-4B6D-BCC4-B2F851254EED. Errno: 2
```
This file exists, but it has an extra `.tmp` extension. It seems to be related to https://github.com/mas-cli/mas/blob/94eb4e5523768bd9902aa1444ade90ee94b17c63/MasKit/Network/NetworkManager.swift#L32

In my case, the error happens every other run. Run 1, no error, and creates the file with `.tmp` extension. Run 2, shows error, and create the file with no extension, which is correct. Run 3, no error, and creates the file with `.tmp` extension... 

## To Reproduce

1. `mas info 1333542190` Run twice

## Expected Behavior

> A clear and concise description of what you expected to happen.

No error..

## Actual Behavior

> A clear and concise description of what actually happened.

Shows error.

## Screenshots, Terminal Output

> If applicable, add screenshots to help explain your problem.

```
$ mas command
...
```

## Additional Context

> Add any other context about the problem here.
> Did it work in a previous version?

There are some discussions on this in https://github.com/mas-cli/mas/issues/306

**Comment by phatblat** [comment]
Thanks for the bug report!

**Comment by chris-araman** [comment]
This should be fixed in #345 and is expected to be included in the next release.
